### ISSUE TITLE
MyData metadata validity facet violates Don't Make Me Think

### ISSUE THREAD

**Issue by pdurbin** [IQSS/dataverse#9836]
I was just playing around with the "Creating datasets with incomplete metadata through API" feature developed in this PR:

- #8940

I'm on Dataverse 5.14.

It's a cool feature, but I think MyData suffers a bit from it.

Imagine the following scenario:

- a user creates a collection (dataverse), a dataset, and uploads a file
- time passes
- the user is looking for their dataset and files

Below is what might happen.

## User clicks MyData 

![Screen Shot 2023-08-28 at 12 37 05 PM](https://github.com/IQSS/dataverse/assets/21006/dc344e84-e527-4237-b93f-6d6d2d4ae2cd)

## User sees "sorry no, results are found"

![Screen Shot 2023-08-28 at 12 37 19 PM](https://github.com/IQSS/dataverse/assets/21006/d636eb7c-5d71-423d-aed1-bc035e93c386)

## Now what? 🤔 

Realistically, at this point the user probably contacts support. Where's my data? Help!

A user with time on their hands and a penchant for clicking things might explore further.

## What if I uncheck "incomplete metadata"?

Nope, nothing.

![Screen Shot 2023-08-28 at 12 39 04 PM](https://github.com/IQSS/dataverse/assets/21006/65716f20-4b36-4ade-beff-a6dda6ed510b)

What if I uncheck "incomplete metadata" AND "valid"?

Finally, we are getting somewhere! There's my dataset and my collection (dataverse)! But where's my file?

![Screen Shot 2023-08-28 at 12 39 14 PM](https://github.com/IQSS/dataverse/assets/21006/c9b0d0e5-4b8e-4b43-83e5-90cbe43744e7)

## Check files to see files

As usual, files are not selected by default but if you check "files" you can see them. This is out of scope for this issue.

![Screen Shot 2023-08-28 at 12 39 19 PM](https://github.com/IQSS/dataverse/assets/21006/f25efeda-afa0-402d-84a9-3527c38ce265)

## In conclusion

When a user visits MyData they should see their data, not "Sorry, no results were found." They shouldn't have to uncheck boxes to see their data.

**Comment by ErykKul** [comment]
@pdurbin, It looks like you are missing the solr field for metadata validity. Updating schema and reindexing should fix it. I hope that the documentation mentions this, at least the release notes should. I have verified this on our pilot, which is on 5.14:

![image](https://github.com/IQSS/dataverse/assets/101262459/f15bcf92-b259-4b01-a12f-7b35ec2a120b)

**Comment by pdurbin** [comment]
@ErykKul I just took a quick look at the PR you opened:

- #9964

This helps explain the behavior I was seeing. Thanks!
